File: source/includes/fts-tutorial/autocomplete/facet.go
package main

import (
	"context"
	"fmt"

	"go.mongodb.org/mongo-driver/v2/bson"
	"go.mongodb.org/mongo-driver/v2/mongo"
	"go.mongodb.org/mongo-driver/v2/mongo/options"
)

func main() {
	// connect to your Atlas cluster
	client, err := mongo.Connect(options.Client().ApplyURI("<connection-string>"))
	if err != nil {
		panic(err)
	}
	defer client.Disconnect(context.TODO())

	// set namespace
	collection := client.Database("sample_mflix").Collection("movies")

	// define pipeline stages
	searchMetaStage := bson.D{{Key: "$searchMeta", Value: bson.M{
		"facet": bson.M{
			"operator": bson.M{
				"autocomplete": bson.M{
					"path": "title", "query": "Gravity",
				},
			},
			"facets": bson.M{
				"titleFacet": bson.M{
					"path": "title", "type": "string",
				},
			},
		},
	}}}

	// run pipeline
	cursor, err := collection.Aggregate(context.TODO(), mongo.Pipeline{searchMetaStage})
	if err != nil {
		panic(err)
	}

	// print results
	var results []bson.D
	if err = cursor.All(context.TODO(), &results); err != nil {
		panic(err)
	}
	for _, result := range results {
		fmt.Println(result)
	}
}

File: content/atlas/source/includes/fts/autocomplete/facet.py
import pymongo

# connect to your Atlas cluster
client = pymongo.MongoClient('<connection-string>')

# define pipeline
pipeline = [{"$searchMeta": { 
    "facet": {
        "operator": {
          "autocomplete": {"path": "title", "query": "Gravity"}
        },
        "facets": {
          "titleFacet": {"type": "string", "path": "title"}
        }}}}]
# run pipeline
result = client["sample_mflix"]["movies"].aggregate(pipeline)

# print results
for i in result:
    print(i)
